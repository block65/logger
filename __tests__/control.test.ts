import { describe, expect, test } from '@jest/globals';
import { testVanillaLogger } from './helpers.js';

describe('Control', () => {
  test('Error Object', async () => {
    const [logger, logPromise] = testVanillaLogger();
    logger.error(Object.assign(new Error('hallo'), { debug: 'wooyeah' }));
    await expect(logPromise).resolves.toMatchSnapshot();
  });

  test('Error Object serialized on non-error level', async () => {
    const [logger, logPromise] = testVanillaLogger();
    logger.info(new Error('hallo'));
    await expect(logPromise).resolves.toMatchSnapshot();
  });
});
