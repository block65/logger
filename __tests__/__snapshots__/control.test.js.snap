// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Control Error Object 1`] = `
Array [
  Object {
    "ctx": Object {
      "hostname": "ws7",
      "pid": 1337,
    },
    "data": Object {
      "err": Object {
        "debug": "wooyeah",
        "message": "hallo",
        "name": "Error",
        "stack": "Error: hallo
    at Object.<anonymous> (/home/mholman/Projects/block65/logger/__tests__/control.test.ts:23:32)
    at Promise.then.completed (/home/mholman/Projects/block65/logger/node_modules/jest-circus/build/utils.js:391:28)
    at new Promise (<anonymous>)
    at callAsyncCircusFn (/home/mholman/Projects/block65/logger/node_modules/jest-circus/build/utils.js:316:10)
    at _callCircusTest (/home/mholman/Projects/block65/logger/node_modules/jest-circus/build/run.js:218:40)
    at _runTest (/home/mholman/Projects/block65/logger/node_modules/jest-circus/build/run.js:155:3)
    at _runTestsForDescribeBlock (/home/mholman/Projects/block65/logger/node_modules/jest-circus/build/run.js:66:9)
    at _runTestsForDescribeBlock (/home/mholman/Projects/block65/logger/node_modules/jest-circus/build/run.js:60:9)
    at run (/home/mholman/Projects/block65/logger/node_modules/jest-circus/build/run.js:25:3)
    at runAndTransformResultsToJestFormat (/home/mholman/Projects/block65/logger/node_modules/jest-circus/build/legacy-code-todo-rewrite/jestAdapterInit.js:170:21)",
      },
    },
    "level": 50,
    "msg": "hallo",
    "time": 2009-02-13T23:31:30.000Z,
  },
]
`;

exports[`Control Error Object serialized on non-error level 1`] = `
Array [
  Object {
    "ctx": Object {
      "hostname": "ws7",
      "pid": 1337,
    },
    "data": Object {
      "err": Object {
        "message": "hallo",
        "name": "Error",
        "stack": "Error: hallo
    at Object.<anonymous> (/home/mholman/Projects/block65/logger/__tests__/control.test.ts:29:17)
    at Promise.then.completed (/home/mholman/Projects/block65/logger/node_modules/jest-circus/build/utils.js:391:28)
    at new Promise (<anonymous>)
    at callAsyncCircusFn (/home/mholman/Projects/block65/logger/node_modules/jest-circus/build/utils.js:316:10)
    at _callCircusTest (/home/mholman/Projects/block65/logger/node_modules/jest-circus/build/run.js:218:40)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at _runTest (/home/mholman/Projects/block65/logger/node_modules/jest-circus/build/run.js:155:3)
    at _runTestsForDescribeBlock (/home/mholman/Projects/block65/logger/node_modules/jest-circus/build/run.js:66:9)
    at _runTestsForDescribeBlock (/home/mholman/Projects/block65/logger/node_modules/jest-circus/build/run.js:60:9)",
      },
    },
    "level": 30,
    "msg": "hallo",
    "time": 2009-02-13T23:31:30.000Z,
  },
]
`;
